src/fabric/mod.rs

```diff
@@ -46,14 +45,18 @@ impl Default for Fabric {
             joints: Vec::new(),
             intervals: HashMap::new(),
             faces: HashMap::new(),
-            push_material: Material {
-                stiffness: 3.0,
-                mass: 1.0,
-            },
-            pull_material: Material {
-                stiffness: 1.0,
-                mass: 0.1,
-            },
+            materials: vec![
+                Material {
+                    role: Push,
+                    stiffness: 3.0,
+                    mass: 1.0,
+                },
+                Material {
+                    role: Pull,
+                    stiffness: 1.0,
+                    mass: 0.1,
+                },
+            ],
```

Comment:
Better to extract this to a const array and call .into() on it here.

```rust
const DEFAULT_MATERIALS: [Material; 2] = [
    Material {
        role: Push,
        stiffness: 3.0,
        mass: 1.0,
    },
    Material {
        role: Pull,
        stiffness: 1.0,
        mass: 0.1,
    },
];
```

Then

```suggestion
            materials: DEFAULT_MATERIALS.into(),
```